sample.go

```diff
@@ -0,0 +1,70 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"os"
+	"strings"
+)
+
+// Sample represents a Google Cloud Platform sample and associated properties.
+type Sample struct {
+	// The Google Cloud Project ID this sample will deploy to
+	googleCloudProject string
+
+	// The local directory this sample is located in
+	dir string
+
+	// The CloudRunService this sample will deploy to
+	cloudRunService *CloudRunService
```

Comment:
nit: (Alot of these names are somewhat long - Go tends to prefer shorter variables names)
```suggestion
	service *service
```

Reply:
+1